### ISSUE TITLE
build fails in EP_vpx

### ISSUE THREAD

**Issue by brianjmurrell** [BelledonneCommunications/linphone-android#517]
```
Scanning dependencies of target EP_vpx
make[3]: Leaving directory '/home/brian/linphone-android/WORK/android-arm64/cmake'
make[3]: Entering directory '/home/brian/linphone-android/WORK/android-arm64/cmake'
[ 45%] Creating directories for 'EP_vpx'
[ 45%] No download step for 'EP_vpx'
[ 46%] No patch step for 'EP_vpx'
[ 46%] No update step for 'EP_vpx'
[ 46%] Performing configure step for 'EP_vpx'
  disabling shared
  enabling error_concealment
  enabling realtime_only
  enabling vp8
  disabling vp9
  disabling install_docs
  disabling debug_libs
  disabling examples
  disabling unit_tests
  disabling tools
Configuring for target 'arm64-android-gcc'
  enabling arm64
  enabling neon
  enabling libyuv
Unable to invoke compiler: /-gcc   --sysroot=/home/brian/android-ndk-r19c/platforms/android-28/arch-arm64 -I/home/brian/android-ndk-r19c//sysroot/usr/include -I/home/brian/android-ndk-r19c//sysroot/usr/include/

Configuration failed. This could reflect a misconfiguration of your
toolchains, improper options selected, or another problem. If you
don't see any useful error messages above, the next step is to look
at the configure error log file (config.log) to determine what
configure was trying to do when it died.
make[3]: *** [CMakeFiles/EP_vpx.dir/build.make:107: /home/brian/linphone-android/WORK/android-arm64/Stamp/EP_vpx/EP_vpx-configure] Error 1
make[3]: Leaving directory '/home/brian/linphone-android/WORK/android-arm64/cmake'
make[2]: *** [CMakeFiles/Makefile2:766: CMakeFiles/EP_vpx.dir/all] Error 2
make[2]: Leaving directory '/home/brian/linphone-android/WORK/android-arm64/cmake'
make[1]: *** [Makefile:84: all] Error 2
make[1]: Leaving directory '/home/brian/linphone-android/WORK/android-arm64/cmake'
make: *** [Makefile:215: arm64-build] Error 2
```

Any ideas?

**Comment by licaon-kter** [comment]
16..19b are supported...not sure about 19c...

**Comment by brianjmurrell** [comment]
So back to #516 where one cannot actually get older releases due to the broken web page.  :-(

**Comment by licaon-kter** [comment]
Did you complain to Google yet?

**Comment by brianjmurrell** [comment]
Heh.  Not yet.

**Comment by brianjmurrell** [comment]
Does this build failure look like it could be related to NDK version though?

**Comment by brianjmurrell** [comment]
Downloading 19b now.

**Comment by brianjmurrell** [comment]
19b doesn't resolve this issue:

```
Scanning dependencies of target EP_vpx
make[3]: Leaving directory '/home/brian/linphone-android/WORK/android-arm64/cmake'
make[3]: Entering directory '/home/brian/linphone-android/WORK/android-arm64/cmake'
[ 35%] Creating directories for 'EP_vpx'
[ 35%] No download step for 'EP_vpx'
[ 36%] No patch step for 'EP_vpx'
[ 36%] No update step for 'EP_vpx'
[ 36%] Performing configure step for 'EP_vpx'
  disabling shared
  enabling error_concealment
  enabling realtime_only
  enabling vp8
  disabling vp9
  disabling install_docs
  disabling debug_libs
  disabling examples
  disabling unit_tests
  disabling tools
Configuring for target 'arm64-android-gcc'
  enabling arm64
  enabling neon
  enabling libyuv
Unable to invoke compiler: /-gcc   --sysroot=/home/brian/android-ndk-r19b/platforms/android-28/arch-arm64 -I/home/brian/android-ndk-r19b//sysroot/usr/include -I/home/brian/android-ndk-r19b//sysroot/usr/include/

Configuration failed. This could reflect a misconfiguration of your
toolchains, improper options selected, or another problem. If you
don't see any useful error messages above, the next step is to look
at the configure error log file (config.log) to determine what
configure was trying to do when it died.
make[3]: *** [CMakeFiles/EP_vpx.dir/build.make:107: /home/brian/linphone-android/WORK/android-arm64/Stamp/EP_vpx/EP_vpx-configure] Error 1
make[3]: Leaving directory '/home/brian/linphone-android/WORK/android-arm64/cmake'
make[2]: *** [CMakeFiles/Makefile2:766: CMakeFiles/EP_vpx.dir/all] Error 2
make[2]: Leaving directory '/home/brian/linphone-android/WORK/android-arm64/cmake'
make[1]: *** [Makefile:84: all] Error 2
make[1]: Leaving directory '/home/brian/linphone-android/WORK/android-arm64/cmake'
make: *** [Makefile:215: arm64-build] Error 2
```

Any further ideas?

**Comment by brianjmurrell** [comment]
Where is the `config.log` that the error output refers to:

```
                                        ...the next step is to look
at the configure error log file (config.log) to determine what
configure was trying to do...
```

**Comment by Viish** [comment]
Each time a new NDK comes out, each time we have to make changes in VPX project to be able to build it...

**Comment by Viish** [comment]
By the way, 19b works just fine but you are on a old version of the submodules.
Take master branch for linphone-sdk repository.

**Comment by brianjmurrell** [comment]
FWIW, i am on the master branch of _linphone-android_.  Shouldn't that have a working linphone-sdk submodule already?  I'm happy to change my submodule.  I just want to sanity check where I'm at.

Alternatively, should I be trying to build something else a bit more stable perhaps?  Hopefully something that builds with NDK 19b, so that I don't have to download yet another one.

**Comment by brianjmurrell** [comment]
I don't actually have a `submodules/linphone-sdk`.  I do have a `submodules/linphone` which is:

```
$ git remote -v
origin	https://gitlab.linphone.org/BC/public/linphone.git (fetch)
origin	https://gitlab.linphone.org/BC/public/linphone.git (push)
```

**Comment by Viish** [comment]
The only think that can be built using NDK 19b is linphone-sdk repository master branch.
linphone-android master is old, it will be soon replaced by feature/release-4.1 branch.

**Comment by Viish** [comment]
linphone-sdk is no longer a submodule.
Take a look at the README file in the feature/release-4.1 branch.

**Comment by brianjmurrell** [comment]
So what is the recommended build branch/tag for a non-linphone developer to build?

**Comment by licaon-kter** [comment]
@brianjmurrell you first build linphone-sdk then bring that AAR to linphone branch features-4.1

FYI: https://gitlab.com/fdroid/fdroiddata/merge_requests/4524/diffs

**Comment by Viish** [comment]
Actually you can bring the AAR to any version of linphone-android you want.

**Comment by brianjmurrell** [comment]
Sorry, what's an AAR?

**Comment by Viish** [comment]
A package that contains the native libraries and the Java wrapper. With that you can focus on building an app, the AAR is just a packaged library.

**Comment by Viish** [comment]
Build with NDK 19c has been fixed in cmake-builder commit 0405cb24859746ff6c9c8e2c3e1c12ec95834c34

**Comment by brianjmurrell** [comment]
Indeed.  I do have a linphone binary built and installed on my phone.  Just investigating how I can confirm it gets push notifications before I go to all of the back-end (SIP server) work.
